### ISSUE TITLE
Full list of known issues in Build #290

### ISSUE THREAD

**Issue by Tyler799** [MachineMuse/MachineMusePowersuits#662]
This is the full list of issues _I_ can find in build #290. 

1: None of the old MPSA materials can be crafted. Their recipes don't exist. They have to be downloaded separately.

2: Even with additional recipes, rubber hoses cannot be crafted. Everything else though apparently can be.

3: When pointing your railgun up, recoil should send you flying down. But it doesn't. It sends you flying _upward_ which is incredibly strange. 

~~4: Shovel doesn't work, period. On any material.~~ Fixed in build 290a. 

~~5: Axe works on logs, but not on any other wooden materials, like chests, wooden staits, etc, etc, etc.~~ Fixed in test build 290a. 

6: The leafblower doesn't correctly detect grass/tallgrass. It detects some botania flowers and all leaves, but no grass, or tall leaves.

**Comment by Xen1290** [comment]
1) you should reset your machinemuse config folder and the new recipes get updated.
2)Same as above
For the others, no clue.
@lehjr @MachineMuse

**Comment by Tyler799** [comment]
When I first downloaded Modular Powersuits build 290 I had installed it completely fresh, it wasn't an update. The config file was fresh, but the recieps still didn't show up. Only after downloading them on the lower half of this page (http://minecraft.curseforge.com/projects/modular-powersuits) could I see the recipes. Even then though, hose didn't show up. Rest worked OK though.

Did those @ symbols act as kinda a PM of sort to get those two to hear about this post?

If so, thanks! Not having a working axe/shovel has really sucked. :P

**Comment by lehjr** [comment]
the site only has 3 recipies, but there are actually 4, one of which adds recipes based on EnderIO items. All 4 can be found in the mod's jar file

the broken shovel and axe is known: https://github.com/MachineMuse/MachineMusePowersuits/issues/628
The test build I mention there uses the same code as in the 290 build except with a material check for the axe and shovel module.

**Comment by Tyler799** [comment]
Ok, so the shovel and axe being broken is known. What test build did you mention? Where can I get it? I'd love to use my axe and shovel again.

**Comment by Tyler799** [comment]
Nevermind, found build 290a. I'll try that until it's officially fixed.

**Comment by eyeonus** [comment]
#3's been around for a long time. I'm guessing it has something to do with how the recoil is calculated.

**Comment by Tyler799** [comment]
Oh, another issue. The leafblower doesn't correctly detect grass/tall grass in FTB Infinity. It detects some Botania flowers (the small ones) and leaves, but that's it. Same issue in build 290A.

**Comment by lehjr** [comment]
To be quite honest, it's impossible to be able to detect every item from every mod.

**Comment by Tyler799** [comment]
But it doesn't detect _vanilla_ grass. All it detects is botania's flowers (not the talls ones). Nothing vanilla works. :c

**Comment by lehjr** [comment]
I'm not sure it's meant to detect grass.
code is here, but I don't see any references to grass:
https://github.com/MachineMuse/MachineMusePowersuits/blob/experimental/src/main/scala/net/machinemuse/powersuits/powermodule/tool/LeafBlowerModule.java

**Comment by Tyler799** [comment]
Hmm. One of the sliders for it is "plant radius" (aside from snow and leaf) and it's always taken out all tall grass in every version before. I mean, before 1.7.10, anyways.

**Comment by lehjr** [comment]
the plant radius and plant loop only covers flowers, why I don't know. 
Original (old) code here:
https://github.com/Andrew2448/Andrew2448PowersuitAddons/blob/master/src/minecraft/andrew/powersuits/modules/LeafBlowerModule.java

Still says flowers,so I dunno. I do recall it seeming to work on more than just leaves though.

**Comment by lehjr** [comment]
I can add grass to it. It's not hard. It really needs to get cleaned up anyway since it already has a separate and almost identical loop for each tip of block it works on.

**Comment by Tyler799** [comment]
Hmm. Weird. Back in 1.6, 1.5, and 1.4 I used it to clear tall grass. Bummer that it isn't intended. It'd be super helpful, though. Trying to build in the plains, want to clear grass. :P

**Comment by Tyler799** [comment]
@lehjr I don't suppose you could add grass, could you? It would really help. :D

**Comment by flowtender** [comment]
1&2) that's a build issue rather than a bug. We can solve that by adding the MPSA component recipes to the core recipes files. We will need balanced recipes for them of course.
3) That's intentional - it was supposed to be like a 'bounce' effect. It should only do that when you're touching the ground, but I might have messed that up somewhere along the line?
6) I guess lehjr is addressing this?

In the future please create separate tickets for each issue! thanks :)

**Comment by lehjr** [comment]
Yeah, the leaf blower needs a bit of work. I really don't think it needs a separate loop for every type of block it handles.

**Comment by flowtender** [comment]
btw lehjr you have been putting an insane amount of work into this; please PM me on IRC or whatever

**Comment by Xen1290** [comment]
Recipes for MPSA components arte already done @MachineMuse

**Comment by lehjr** [comment]
Recipes may not be auto extracting over old ones. To me, that would be intended behavior because if you set up a custom recipe you wouldn't want it overwritten, though some mods to have a separate "custom recipe" handling.

**Comment by flowtender** [comment]
oh, that's right

**Comment by Tyler799** [comment]
Hey all!

Just noticed build #291. :D   

The issue with the railgun is mid-air. If I'm in mid air and point down, I go up, as expected. But if I point up, instead of going downward it launches me _upward_ which is really strange.

Also perhaps a module to reduce railgun recoil? I'd be totally cool with it being really heavy. I have like 21kg to play with. :)

**Comment by lehjr** [comment]
something like an anchor and a chain?  :P

**Comment by flowtender** [comment]
Railgun recoil is in fact reduced by weight if you have a lot of it ;)

**Comment by flowtender** [comment]
Can we please get separate tickets for each of the issues here that are still unfixed? Thanks!
